src/app/Shared/Context/FilterContext.tsx

```diff
@@ -1,19 +1,20 @@
 import React, { useState, createContext } from "react";
-import { FilterInterface, mainFilterAttributes } from "../Model/Filter.model";
+import { FilterInterface } from "../Model/Filter.model";
 
-const generateEmptyFilter = (): FilterInterface => {
-    const defaultFilter = {} as FilterInterface;
-    mainFilterAttributes.forEach((element) => {
-        defaultFilter[element] = [];
-    });
-    return defaultFilter;
-};
-
-export const defaultFilter = generateEmptyFilter();
+interface FilterContextInterface {
+    mainFilter: string[];
+    selectedFilter: FilterInterface;
+    emptyFilter: FilterInterface;
+}
 
+const defaultFilter: FilterContextInterface = {
+    mainFilter: [],
+    selectedFilter: {},
```

Comment:
Consider using null if you want to express that something is explicitly not set. {} is used more for things that are yet uninitialized. Then you can also express a not set or empty state in the interface like selectedFilter: FilterInterface | null.

Reply:
if your semantic is that selectedFilter defaults to an empty collection of filters you can forget the previous comment. But you should perhaps consider using an array or a set then.

Reply:
done too, i think